templates/repo/diff/box.tmpl

```diff
@@ -15,11 +15,17 @@
 	<div>
 		<div class="diff-detail-box diff-box sticky gt-df gt-sb gt-ac gt-fw">
 			<div class="gt-df gt-ac gt-fw">
-				<a class="diff-toggle-file-tree-button muted gt-df gt-ac">
+				<button class="diff-toggle-file-tree-button gt-df gt-ac">
 					{{/* the icon meaning is reversed here, "octicon-sidebar-collapse" means show the file tree */}}
-					{{svg "octicon-sidebar-collapse" 20 "icon hide"}}
-					{{svg "octicon-sidebar-expand" 20 "icon"}}
-				</a>
+					{{svg "octicon-sidebar-collapse" 20 "icon gt-hidden"}}
+					{{svg "octicon-sidebar-expand" 20 "icon gt-hidden"}}
+				</button>
+				<script>
+					const treeVisible = localStorage?.getItem('diff_file_tree_visible') === 'true';
+					const btn = document.querySelector('.diff-toggle-file-tree-button');
+					const icon = `.octicon-sidebar-${treeVisible ? 'expand' : 'collapse'}`;
```

Comment:
These lines will case: `window.btn`, `window.icon`, etc

Reply:
How to avoid? I can't set `type=module` unfortunately as I need synchronous execution.

Reply:
Does not seem to be the case at least in Firefox:

<img width="172" alt="image" src="https://user-images.githubusercontent.com/115237/226631855-10ce9efa-10db-49f1-87ec-c90eeb98d3b1.png">

Reply:
Maybe comment + prefixWithLongEnoughName to avoid conflicts, or CustomElements.

In history, various kinds of pollution already caused a lot of problems.

Reply:
CustomElements does not work easily in this case as I need access to the svg data, which means I'd have to import `svg.js` into the js chunk which would considerably blow it up in size and duplicate the SVG data into it.

Reply:
Added variable prefixes.